src/SleepRepository.js

```diff
@@ -10,19 +10,26 @@ class SleepRepository {
   }
 
   getAllTimeAvg() {
-    const totalHrs = this.user.reduce((totalHours, day) => {
-      totalHours += day.hoursSlept;
-      return totalHours;
-    }, 0);
-    return Math.round(totalHrs / this.user.length);
+    const total = this.getTotalOf('hoursSlept', 'user')
+    return parseFloat((total / this.user.length).toFixed(1));
   }
 
   getQualitySleepAvg() {
-    const sleepQualAvg = this.user.reduce((sleepQual, day) => {
-      sleepQual += day.sleepQuality;
-      return sleepQual;
+    const total = this.getTotalOf('sleepQuality', 'user')
+    return parseFloat((total / this.user.length).toFixed(1));
+  }
+
+  getAvgQuality() {
+    const total = this.getTotalOf('sleepQuality', 'sleepData')
+    return parseFloat((total / this.sleepData.length).toFixed(1));
+  }
+
+  getTotalOf(metric, datatype) {
```

Comment:
We could even have functions like this that are general all be in the UserRepo class, that way all the class methods could reference that, instead of being a method in each one.

Reply:
That's an interesting Idea! I know each class needs to get access to averages data - maybe this is how we solve that issue!